src/se/simple_type.rs

```diff
@@ -1219,4 +1288,44 @@ mod tests {
             assert_eq!(buffer, "1 2 3");
         }
     }
+
+    mod cdata {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        macro_rules! serialize_cdata_as {
+            ($name:ident: $data:expr => $expected:literal) => {
+                #[test]
+                fn $name() {
+                    let ser = SimpleTypeSerializer {
+                        writer: String::new(),
+                        target: QuoteTarget::Text,
+                        level: QuoteLevel::Full,
+                        format: TextFormat::CData,
+                    };
+
+                    let buffer = $data.serialize(ser).unwrap();
+                    assert_eq!(buffer, $expected);
+                }
+            };
+        }
+
+        serialize_cdata_as!(empty_string: "" => "");
+        serialize_cdata_as!(simple_text: "Hello World" => "<![CDATA[Hello World]]>");
+        serialize_cdata_as!(with_markup: "<tag>content</tag>" => "<![CDATA[<tag>content</tag>]]>");
+        serialize_cdata_as!(with_ampersand: "Tom & Jerry" => "<![CDATA[Tom & Jerry]]>");
+        serialize_cdata_as!(with_quotes: r#"He said "Hello""# => r#"<![CDATA[He said "Hello"]]>"#);
+        serialize_cdata_as!(all_xml_chars: "<>&\"'" => "<![CDATA[<>&\"']]>");
+
+        serialize_cdata_as!(with_cdata_end: "foo]]>bar" => "<![CDATA[foo]]]]><![CDATA[>bar]]>");
+        serialize_cdata_as!(multiple_cdata_ends: "a]]>b]]>c" => "<![CDATA[a]]]]><![CDATA[>b]]]]><![CDATA[>c]]>");
+        serialize_cdata_as!(starts_with_cdata_end: "]]>hello" => "<![CDATA[]]]]><![CDATA[>hello]]>");
+        serialize_cdata_as!(ends_with_cdata_end: "hello]]>" => "<![CDATA[hello]]]]><![CDATA[>]]>");
+        serialize_cdata_as!(only_cdata_end: "]]>" => "<![CDATA[]]]]><![CDATA[>]]>");
+
+        serialize_cdata_as!(seq_basic: vec!["foo", "bar", "baz"] => "<![CDATA[foo]]> <![CDATA[bar]]> <![CDATA[baz]]>");
+        serialize_cdata_as!(seq_with_markup_chars: vec!["<tag>", "&entity", "\"quoted\"", "with space"] => "<![CDATA[<tag>]]> <![CDATA[&entity]]> <![CDATA[\"quoted\"]]> <![CDATA[with space]]>");
+        serialize_cdata_as!(seq_with_cdata_end_split: vec!["foo]]>bar", "test"] => "<![CDATA[foo]]]]><![CDATA[>bar]]> <![CDATA[test]]>");
+        serialize_cdata_as!(tuple_cdata: ("first", 42, "third") => "<![CDATA[first]]> <![CDATA[42]]> <![CDATA[third]]>");
```

Comment:
It would be better, if this serialized as one CDATA section. Probably, that is possible?

Reply:
yeah agreed, need to figure out exactly how to do that nicely with serde